### PR TITLE
Feat helm unittest

### LINKED ISSUE #239: Unit tests
It would be nice to test out the helm chart using unit tests.

This has come up a few times since contributions can introduce breaking changes.

Options

1. Using https://www.phillipsj.net/posts/yes-you-can-unit-test-helm-charts/ with golang and terratest
2. Using https://blog.ediri.io/how-to-unit-test-your-helm-charts with [quintush/helm-unittest](https://github.com/quintush/helm-unittest)
3. helm test
    - The command `helm test` is more of an integration test since it requires deploying the chart 

Whatever is chosen, the tests should capture at least

- enabling a single vcs - test github, gitlab, bitbucket, azuredevops, etc
- tbd

### PR DESCRIPTION
**Pull request by roytev**
## what

- Add unit tests to the helm chart with all of the values available.

## why

- Make the chart more robust.
- Allow contributors to be more safe when introduce new features to the chart

## references

-  closes #239

### CODE REVIEW

File: charts/atlantis/tests/secret-aws_test.yaml
Comment (nitrocode) [inline charts/atlantis/tests/secret-aws_test.yaml:1]: This is fantastic work. You beat me to it!

File: charts/atlantis/templates/secret-webhook.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ data:
   key.pem: {{ required "githubApp.key is required if githubApp configuration is specified." .Values.githubApp.key | b64enc }}
   github_secret: {{ required "githubApp.secret is required if githubApp configuration is specified." .Values.githubApp.secret | b64enc }}
   {{- end}}
-  {{- if .Values.github.user }}
```
Comment (GMartinez-Sisti) [inline charts/atlantis/templates/secret-webhook.yaml:13]: While this makes sense, I would change this on a different PR. Also since there are other places where this can be improved ([example](https://github.com/runatlantis/helm-charts/blob/main/charts/atlantis/templates/statefulset.yaml#L296)).

File: charts/atlantis/values.schema.json [hunk 2]
```diff
@@ -35,6 +35,14 @@
...
       ]
     },
+    "orgWhitelist":{
+      "type":"string",
+      "default":"<replace-me>",
+      "description":"Deprecated (see orgAllowlist) List of repositories from which Atlantis will accept webhooks. Accepts wildcard characters (`*`). Multiple values may be comma-separated.",
+      "examples":[
+        "github.com/myorg/*"
+      ]
+    },
```
Comment (GMartinez-Sisti) [inline charts/atlantis/values.schema.json:45]: This is another example of a change that would be better in a different PR 😃

**Review by GMartinez-Sisti — APPROVED** [review]
Looks awesome! Some minor remarks 😃

File: charts/atlantis/tests/secret-aws_test.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+suite: test secret-aws for aws
```
[the remark this answers is not in the document]
Comment (GMartinez-Sisti) [reply charts/atlantis/tests/secret-aws_test.yaml:2]: If this job is added, then it needs to be removed from `ct lint` https://github.com/runatlantis/helm-charts/pull/334/files#diff-a659d2a19976a325533dea0495eed87496438b045319fd292efca590796060e1R7-R8

File: charts/atlantis/templates/secret-webhook.yaml (hunk 1 above)
Comment (roytev) [reply charts/atlantis/templates/secret-webhook.yaml]: Removed it will be taken in another pr

**Review by GenPage — APPROVED** [review]
